genie-web/src/main/java/com/netflix/genie/web/util/BaseRegexAgentMetadataInspector.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.web.util;
+
+import com.netflix.genie.web.services.AgentFilterService;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Abstract base class for {@link AgentFilterService.AgentMetadataInspector} based on regex matching.
+ * Can be configure to accept or reject in case of match. It is a pass-through otherwise.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+abstract class BaseRegexAgentMetadataInspector implements AgentFilterService.AgentMetadataInspector {
+
+    private final AtomicReference<Pattern> patterncReference = new AtomicReference<>();
+    private final AgentFilterService.InspectionReport.InspectionDecision decisionIfMatch;
+    private String lastCompiledPattern;
+
+    BaseRegexAgentMetadataInspector(
+        final AgentFilterService.InspectionReport.InspectionDecision decisionIfMatch
```

Comment:
Blargh. Maybe based on this you should break `InspectionReport` into its own class file? Pretty long coordinate